xdsl/backend/riscv/lowering/rv32_arith_lowering.py

```diff
@@ -0,0 +1,436 @@
+                flt2 = riscv.FltSOP(op.rhs, op.lhs)
+                rewriter.replace_matched_op([flt1, flt2, riscv.OrOp(flt2, flt1)])
+            # ord
+            case 7:
+                feq1 = riscv.FeqSOP(op.lhs, op.lhs)
+                feq2 = riscv.FeqSOP(op.rhs, op.rhs)
+                rewriter.replace_matched_op([feq1, feq2, riscv.AndOp(feq2, feq1)])
+            # ueq
+            case 8:
+                flt1 = riscv.FltSOP(op.lhs, op.rhs)
+                flt2 = riscv.FltSOP(op.rhs, op.lhs)
+                or_ = riscv.OrOp(flt2, flt1)
+                rewriter.replace_matched_op([flt1, flt2, or_, riscv.XoriOp(or_, 1)])
+            # ugt
+            case 9:
+                fle = riscv.FleSOP(op.lhs, op.rhs)
+                rewriter.replace_matched_op([fle, riscv.XoriOp(fle, 1)])
+            # uge
+            case 10:
+                fle = riscv.FltSOP(op.lhs, op.rhs)
+                rewriter.replace_matched_op([fle, riscv.XoriOp(fle, 1)])
+            # ult
+            case 11:
+                fle = riscv.FleSOP(op.rhs, op.lhs)
+                rewriter.replace_matched_op([fle, riscv.XoriOp(fle, 1)])
+            # ule
+            case 12:
+                flt = riscv.FltSOP(op.rhs, op.lhs)
+                rewriter.replace_matched_op([flt, riscv.XoriOp(flt, 1)])
+            # une
+            case 13:
+                feq = riscv.FeqSOP(op.lhs, op.rhs)
+                rewriter.replace_matched_op([feq, riscv.XoriOp(feq, 1)])
+            # uno
+            case 14:
+                feq1 = riscv.FeqSOP(op.lhs, op.lhs)
+                feq2 = riscv.FeqSOP(op.rhs, op.rhs)
+                and_ = riscv.AndOp(feq2, feq1)
+                rewriter.replace_matched_op([feq1, feq2, and_, riscv.XoriOp(and_, 1)])
+            # true
+            case 15:
+                rewriter.replace_matched_op([riscv.LiOp(1)])
+            case _:
+                raise NotImplementedError("Cmpf predicate not supported")
+
+
+class LowerArithSIToFPOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: arith.SIToFPOp, rewriter: PatternRewriter) -> None:
+        rewriter.replace_matched_op([riscv.FCvtSWOp(op.input)])
+
+
+class LowerArithFPToSIOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: arith.FPToSIOp, rewriter: PatternRewriter) -> None:
+        rewriter.replace_matched_op([riscv.FCvtWSOp(op.input)])
+
+
+class LowerArithExtFOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: arith.ExtFOp, rewriter: PatternRewriter) -> None:
+        raise NotImplementedError("ExtF is not supported")
+
+
+class LowerArithTruncFOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: arith.TruncFOp, rewriter: PatternRewriter) -> None:
+        raise NotImplementedError("TruncF is not supported")
+
+
+class RISCVLowerArithRV32(ModulePass):
+    name = "riscv-lower-arith-rv32"
```

Comment:
Is there a canonical way to name conversion passes? I would have named this `lower-arith-to-rv32` or `convert-arith-to-rv32`. Not sure if this is right, but this order in naming looks a bit redundant and backwards.

Reply:
That's a good point. I think `arith-to-riscv` or `lower-arith-to-riscv` are good. not sure we want to specify `rv32`, we haven't so far, might want to stay consistent and rename things later